### ISSUE TITLE
Notification deleteIntent

### ISSUE THREAD

**Issue by opusforlife2** [TeamNewPipe/NewPipe#4335]
Reminder for @Stypox.

**Comment by starboi02** [comment]
Can I work on this issue?

**Comment by opusforlife2** [comment]
This feature needs some groundwork first, as written in the OP. Are you wanting to work on that or deleteIntent itself?

**Comment by starboi02** [comment]
I would like to give it a try , can easily do the deleteIntent part though but need to work on the prerequisite one.

**Comment by Stypox** [comment]
@starboi02 in order to implement queue saving, you have to save it to the database. Theoretically you could just add a boolean column to the `streams` table representing whether the stream is saved or not, but this approach is not future-proof, since sooner or later we may want to allow the user to save multiple play queues, if he wants to. Hence, I would build the system in such a way that we can expand it in the future, thus allowing to store multiple queues. This is what I'd do:
- Create a `queues` table, keeping in mind that queues could have continuations, represented using the [`Page`](https://github.com/TeamNewPipe/NewPipeExtractor/blob/dev/extractor/src/main/java/org/schabi/newpipe/extractor/Page.java) class and the [`AbstractInfoPlayQueue`](https://github.com/TeamNewPipe/NewPipe/blob/87228673b45225a57b53937ab2c9715d7bf205ca/app/src/main/java/org/schabi/newpipe/player/playqueue/AbstractInfoPlayQueue.java) class (e.g. when you play a playlist, it is not downloaded as a whole at once, but it is instead downloaded piece by piece on request). So these would be the columns :
  - `primaryKey`: the autoincrement integer uniquely representing the queue
  - `queueIndex`: nonnull integer representing the queue integer
  - `serviceId`: nullable integer: if present, indicates this queue is an `AbstractInfoPlayQueue`; represents the continuations service id for `AbstractInfoPlayQueue`
  - `baseUrl`: nullable string: `AbstractInfoPlayQueue`'s `baseUrl`
  - `nextPageUrl`: nullable string: `AbstractInfoPlayQueue.nextPage`'s `url`
  - `nextPageId`: nullable string: `AbstractInfoPlayQueue.nextPage`'s `id`
  - <sub>(`AbstractInfoPlayQueue.nextPage`'s `ids`, is not here because it requires a `1 to N` relationship, see below)</sub>
  - `pageCookies`: nullable string representing comma-separated cookies from `AbstractInfoPlayQueue.nextPage`'s `cookies` (like in the browser)
- Create a `queue_page_ids` table, representing a `1 to N` relationship to the `queues` table with `AbstractInfoPlayQueue.nextPage`'s `ids`, made of two columns:
  - `queue_pk`: nonnull integer: the queue primary key this id belongs to
  - `page_id`: nonnull string: the id from the page
- Create a `queue_items` table, representing a `N to N` relationship between the `streams` table and the `queues` one. Therefore it should have two integer columns:
  - `stream_pk`: nonnull integer: the stream primary key
  - `queue_id`: nonnull integer: the queue primary key
- Change `StreamDAO.deleteOrphans()` so that it makes sure the stream is not used inside `queue_items` before deleting it from the database.

With such an infrastructure in place you should be able to save the playqueue on player shutdown (but not when the user manually closes the player) and restore it when the user taps play again in the notification. If you have any question feel free to ask ;-)

**Comment by opusforlife2** [comment]
:O

**Comment by opusforlife2** [comment]
@starboi02 Do you wish to tackle this? If so, I'll assign you this issue. If not, please respond so someone else is free to work on this.

**Comment by starboi02** [comment]
@opusforlife2 I tried making things work but its not going the way i expected, you can assign anyone who's willing to work on this.

**Comment by opusforlife2** [comment]
> but its not going the way i expected

It's fine if it takes you time to figure things out, you know. I just need to know if you're willing to work on it and submit a PR eventually.

**Comment by starboi02** [comment]
@opusforlife2 yes I am willing to work on this and would eventually try to submit a PR.

**Comment by opusforlife2** [comment]
Awesome! Thank you for taking this on. :)

**Comment by opusforlife2** [comment]
@starboi02 Hey. What's the status? Any progress?

**Comment by SameenAhnaf** [comment]
@opusforlife2 I think, this functionality is already included in the app. Could you please check and confirm it??

**Comment by Stypox** [comment]
@SameenAhnaf no, this is not included, you still have to press "X" to stop the player, and swiping the notification is not doable

**Comment by SameenAhnaf** [comment]
@Stypox No matter what I do, it's swippable on my device. 🙃


https://user-images.githubusercontent.com/79245641/154472010-d837f987-a7ef-40e5-82c0-627082977921.mp4

**Comment by TiA4f8R** [comment]
It's a MIUI issue because you can swipe the player but it's still playing the content (basing on the screen recording you provided) and is not the behavior excepted.

###### Another bad MIUI customization of Android...

**Comment by XiangRongLin** [comment]
@Stypox Are you still working on this, since you are assigned

**Comment by opusforlife2** [comment]
Traitor.
